server/src/main/java/org/elasticsearch/rest/RestUtils.java

```diff
@@ -280,4 +288,28 @@ public static TimeValue getMasterNodeTimeout(RestRequest restRequest) {
         return restRequest.paramAsTime(REST_MASTER_TIMEOUT_PARAM, REST_MASTER_TIMEOUT_DEFAULT);
     }
 
+    /**
+     * Extract the {@code ?master_timeout} parameter from the request, imposing the common default of {@code 30s} in case the parameter is
+     * missing.
+     *
+     * @param restRequest The request from which to extract the {@code ?timeout} parameter
+     * @return the timeout from the request, with a default of {@link AcknowledgedRequest#DEFAULT_ACK_TIMEOUT} ({@code 30s}) if the request
+     *         does not specify the parameter
+     */
+    public static TimeValue getAckTimeout(RestRequest restRequest) {
+        assert restRequest != null;
+        return restRequest.paramAsTime(REST_TIMEOUT_PARAM, DEFAULT_ACK_TIMEOUT);
```

Comment:
This may be controversial, in #107986 a constant was added to `RestUtils`, rather than using the constant defined in `MasterNodeRequest`. I'm open to the alternative of creating a local constant and deprecating the one in `AcknowledgedRequest` if we think that's a better approach.

Reply:
We can reuse this constant from `AcknowledgedRequest` for now. I suspect we will want to change it eventually for similar reason that led to creation of `REST_MASTER_TIMEOUT_DEFAULT` here. But this is _not_ as critical as `master_timeout` since the cluster state update will still be applied even if it times out. So we can tackle it in future.